src/evo/mnauth.cpp

```diff
@@ -114,23 +114,33 @@ void CMNAuth::ProcessMessage(CNode* pnode, const std::string& strCommand, CDataS
     }
 }
 
-void CMNAuth::NotifyMasternodeListChanged(const CDeterministicMNList& newList)
+void CMNAuth::NotifyMasternodeListChanged(bool undo, const CDeterministicMNList& oldMNList, const CDeterministicMNListDiff& diff)
```

Comment:
Yepp, it is for now. In one of my early versions of DIP3, I used the same notification system with `oldList` and `diff`, but then ditched it completely because I didn't like that a consumer of the signal has no way to differentiate between real changes to the list and reverting of the list due to reorgs. My solution to fix this (theoretical) problem was now to introduce the `undo` flag, which at least gives the possibility to differentiate and kind of forces future consumers of the signal to think about that flag and the implications of it. It's like the `fInitialDownload` in the `UpdatedBlockTip`...even if it wouldn't be used right now, it would still force us into thinking about it each time we implement a consumer of the signal.